Training/Program.cs

```diff
@@ -1,33 +1,82 @@
 ﻿// ------------------------------------------------------------------------------------------------
 // Training ~ A training program for new joinees at Metamation, Batch- July 2023.
-// Copyright (c) Metamation India.                                                
+// Copyright (c) Metamation India.
 // ------------------------------------------------------------------
-// Program.cs                                                                     
-// Program on main branch.
+// Program.cs
+// Program to implement double.Parse.
 // ------------------------------------------------------------------------------------------------
-namespace Training {
-   #region class Program --------------------------------------------------------------------------
-   internal class Program {
-      #region Constructors ------------------------------------------
-      #endregion
+using static System.Console;
+namespace Training;
 
-      #region Method ------------------------------------------------
-      static void Main (string[] args) {
-         Console.WriteLine ("Hello, World!");
-      }
-      #endregion
-
-      #region Operators ---------------------------------------------
-      #endregion
+#region class Program -----------------------------------------------------------------------------
+internal class Program {
+   #region Method ---------------------------------------------------
+   static void Main (string[] args) {
+      Write ("input:");
+      string input = ReadLine ().Trim ().ToLower () ?? "";
+      bool isValid = DoubleParse.Parse (input, out double num);
+      WriteLine ("output: " + num + $"\n{(isValid ? "" : "invalid input")}");
+   }
+   #endregion
+}
+#endregion
 
-      #region Implementation ----------------------------------------
-      #endregion
+#region Class DoubleParse -------------------------------------------------------------------------
+public class DoubleParse {
+   #region Method ---------------------------------------------------
+   /// <summary>Converts string to double.</summary>
+   /// <param name="input">string input</param>
+   /// <param name="num">double number</param>
+   /// <returns>If False is input number is invalid, true otherwise.</returns>
+   public static bool Parse (string input, out double num) {
+      num = 0;
+      if (input != "") {
+         string[] number = input.Split ('e');
+         if (number.Length > 2) return false;
+         if (Validate (number[0] ?? "", out bool isNegative)) {
+            num = Digits (number[0], isNegative);
+            if (number.Length == 2)
+               if (Validate (number[1], out isNegative)) {
+                  double exponent = Digits (number[1], isNegative);
+                  num *= Math.Pow (10, exponent);
+               } else {
+                  num = 0;
+                  return false;
+               }
+            return true;
+         }
+      }

```

Comment:
See [guard](https://en.wikipedia.org/wiki/Guard_(computer_science))
```suggestion
     if (intput == "") return false;
         string[] number = input.Split ('e');
         if (number.Length > 2) return false;
         if (Validate (number[0] ?? "", out bool isNegative)) {
            num = Digits (number[0], isNegative);
            if (number.Length == 2)
               if (Validate (number[1], out isNegative)) {
                  double exponent = Digits (number[1], isNegative);
                  num *= Math.Pow (10, exponent);
               } else {
                  num = 0;
                  return false;
               }
            return true;
         }

```